backend/src/Designer/EventHandlers/LanguageTextsKeyChanged/LanguageTextsIdChanged.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Events;
+using Altinn.Studio.Designer.Hubs.SyncHub;
+using Altinn.Studio.Designer.Services.Interfaces;
+using MediatR;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Altinn.Studio.Designer.EventHandlers.LanguageTextsKeyChanged;
+
+public class LanguageTextsIdChanged(
+    IHubContext<SyncHub, ISyncClient> syncHub, ITextsService textsService) : INotificationHandler<LanguageTextsKeyChangedEvent>
+{
+
+    public async Task Handle(LanguageTextsKeyChangedEvent notification, CancellationToken cancellationToken)
+    {
+        List<string> updatedFiles = await textsService.UpdateRelatedFiles(notification.EditingContext.Org, notification.EditingContext.Repo, notification.EditingContext.Developer, notification.idMutations);
```

Comment:
I also wonder if we keep all layouts in the same layoutset in the same cache in frontend so we should instead invalidate the layoutset?

Reply:
See this file
https://github.com/Altinn/altinn-studio/blob/main/frontend/packages/shared/src/queryInvalidator/SyncSuccessQueriesInvalidator.ts#L33

Reply:
I'm wondering if a better solution to solve this is to make the frontend decide what caches to invalidate based on which files were changed? That way the backend doesn't need to think about how the frontend cache logic works.

For example if `QueryKey.FormLayouts` should be invalidated for all layout changes, then  matching `fileSyncSuccess` messages with a regex like `^App/ui/.*/layouts.*` could possibly work. Maybe i am missing some part of the idea behind the fileSync websocket messages here?

Reply:
Yes, that could be a way to handle it. But then we are sending unnecessary meny syncSuccesses to the frontend tho. And the frontend will have to "calculate" what layoutset the layout in the syncsuccess is belonging to. 

So if you think that is a better way to solve it we need to adapt the invalication logic in frontend and we also need to adapt the syncsuccess-responses from the componentIDEvent